### PR TITLE
Don't merge - netdb WIP

### PR DESCRIPTION
**Pull request by sajattack**
Just trying to get a comparison to see how I broke dns

### CODE REVIEW

File: newlib/libc/sys/redox/src/netdb.rs [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+                    q_class: 0x0001,
+                },
+            ],
+            answers: vec![],
+        };
+
+        let packet_data = packet.compile();
+
+        let fd = ::RawFile::open(
+            format!("udp:{}.{}.{}.{}:0", ip[0], ip[1], ip[2], ip[3]).as_bytes(),
```
Comment (jackpot51) [inline newlib/libc/sys/redox/src/netdb.rs]: This is missing a slash, it should be `udp:/{}.{}.{}.{}:0`

**Comment by jackpot51** [comment]
Also, `inet_ntoa` should be `char *inet_ntoa(struct in_addr in);`, not `char *inet_ntoa(struct * in_addr in);`

**Comment by sajattack** [comment]
So add an ``&``` at line 259?

**Comment by jackpot51** [comment]
I merged the changes after fixing them, please sync with the redox branch

**Comment by jackpot51** [comment]
I think I messed that up, I will revert.

File: newlib/libc/sys/redox/src/netdb.rs [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+                Ok(LookupHost(addrs.into_iter()))
+            }
+            Err(_err) => Err(Error::new(EINVAL)),
+        }
+    } else {
+        Err(Error::new(EINVAL))
+    }
+}
+
+unsafe fn lookup_addr(addr: *const in_addr) -> Result<LookupAddr> {
```
Comment (jackpot51) [inline newlib/libc/sys/redox/src/netdb.rs]: This should be `addr: in_addr`

File: newlib/libc/sys/redox/src/netdb.rs [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+
+libc_fn!(unsafe endservent() {
+    let _ = syscall::close(SERVDB);
+});
+
+
+libc_fn!(unsafe gethostbyaddr(v: *const libc::c_void, length: socklen_t, format: libc::c_int) -> Result <*const hostent> {
+    let mut addr = mem::uninitialized(); 
+    ::socket::inet_aton(['8' as i8, '.' as i8, '8' as i8, '.' as i8, '4' as i8, '.' as i8, '4' as i8, '\0' as i8].as_ptr(), &mut addr);
+    match lookup_addr(&addr) {
```
Comment (jackpot51) [inline newlib/libc/sys/redox/src/netdb.rs]: This should be `lookup_addr(addr)`

**Comment by jackpot51** [comment]
I fixed `inet_ntoa` in the redox branch. Please rebase

**Pull request closed without merging**